### ISSUE TITLE
Crash because of NULL pointer for input event

### ISSUE THREAD

**Issue by toger5** [godotengine/godot#8858]
In [Input_event line 60](https://github.com/godotengine/godot/blob/5b3709d3096df737b8bb2344446be818b0389bfe/core/os/input_event.cpp#L60) the passed Input event is NULL although `this` is a valid input event object.

``` c++
bool InputEvent::is_action(const StringName &p_action) const {

	return InputMap::get_singleton()->event_is_action(Ref<InputEvent>(this), p_action);
}
```
So there is something wrong with this:
```c++
Ref<InputEvent>(this)
```
I think it is an easy fix, I can also do it if someone tells me what the problem is.

**Comment by toger5** [comment]
does it maybe has something to do with [this](https://github.com/godotengine/godot/blob/5b3709d3096df737b8bb2344446be818b0389bfe/platform/x11/os_x11.cpp#L1160):
```c++
//don't set mod state if modifier keys are released by themselves
	//else event.is_action() will not work correctly here
	if (!k->is_pressed()) {
		if (k->get_scancode() == KEY_SHIFT)
			k->set_shift(false);
		else if (k->get_scancode() == KEY_CONTROL)
			k->set_control(false);
		else if (k->get_scancode() == KEY_ALT)
			k->set_alt(false);
		else if (k->get_scancode() == KEY_META)
			k->set_metakey(false);
	}
```

**Comment by akien-mga** [comment]
Fixed by 6161e73.
